lms/djangoapps/course_api/blocks/transformers/extra_fields.py

```diff
@@ -0,0 +1,71 @@
+"""
+Extra Fields Transformer
+"""
+import re
+
+from django.conf import settings
+
+from openedx.core.djangoapps.content.block_structure.transformer import BlockStructureTransformer
+
+
+class ExtraFieldsTransformer(BlockStructureTransformer):
+    """
+    Shows ExtraFields specified in COURSE_BLOCKS_API_EXTRA_FIELDS LMS Django Settings variable
+    """
+    WRITE_VERSION = 1
+    READ_VERSION = 1
+    EXTRA_FIELDS_DATA = 'extra_fields'
+
+    @classmethod
+    def name(cls):
+        return "blocks_api:extra_fields"
+
+    @classmethod
+    def collect(cls, block_structure):
+        """
+        Collects any information that's necessary to execute this transformer's
+        transform method.
+        """
+        requested_field_names = cls.get_requested_extra_fields()
+
+        block_structure.request_xblock_fields('category', *requested_field_names)
+
+    @classmethod
+    def get_requested_extra_fields(cls):
+        """
+        Returns the names of the requested extra fields
+        """
+        if settings.FEATURES.get('COURSE_BLOCKS_API_EXTRA_FIELDS') is None:
+            return []
+
+        return [field_name for block_type, field_name in settings.FEATURES.get('COURSE_BLOCKS_API_EXTRA_FIELDS')]
+
+    def transform(self, usage_info, block_structure):
+        """
+        Mutates block_structure based on the given usage_info.
+        """
+        if len(self.get_requested_extra_fields()) == 0:
+            return
+
+        requested_fields = [dict(
+            block_type=block_type,
+            field_name=field_name,
+        ) for block_type, field_name in settings.FEATURES.get('COURSE_BLOCKS_API_EXTRA_FIELDS')]
+
+        for block_key in block_structure.topological_traversal():
+            for field in requested_fields:
```

Comment:
Nit: I think it'd be easier to follow if you rename this ambiguous variable `field` to `requested_field`